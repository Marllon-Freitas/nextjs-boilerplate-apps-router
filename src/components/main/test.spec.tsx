import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // renderizar o componente 'Main'
    const { container } = render(<Main />)
    // busca o elemento e verifica se ele existe
    expect(
      screen.getByRole('heading', { name: /hello world/i })
    ).toBeInTheDocument()

    // gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    // renderizar o componente 'Main'
    const { container } = render(<Main />)
    // busca o elemento e verifica se background-color Ã© correto
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
