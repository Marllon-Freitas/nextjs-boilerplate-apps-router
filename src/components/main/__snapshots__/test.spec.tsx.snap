// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #f2f2f2;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

<main
  class="c0"
>
  <h1>
    Hello world
  </h1>
</main>
`;
